src/machine/machine_mimxrt1062.go

```diff
@@ -301,6 +330,101 @@ func (p Pin) Toggle() {
 	gpio.DR_TOGGLE.Set(p.getMask())
 }
 
+// dispatchInterrupt invokes the user-provided callback functions for external
+// interrupts generated on the high-speed GPIO pins.
+//
+// Unfortunately, all four high-speed GPIO ports (A-D) are connected to just a
+// single interrupt control line. Therefore, the interrupt status register (ISR)
+// must be checked in all four GPIO ports on every interrupt.
+func (jt *pinJumpTable) dispatchInterrupt(interrupt.Interrupt) {
+	handle := func(gpio *nxp.GPIO_Type, port Pin) {
```

Comment:
This is probably okay but note that closures might cause a heap allocation, which is not allowed in interrupts (see #1460).

Reply:
Hmm, why is this okay then? I can verify it works, and it is an interrupt service routine. It _looks like_ a closure over the receiver variable `jt`... otherwise, maybe it is just an anonymous function? 

Of course I can refactor and pull this out as a proper function if you recommend it

Reply:
This is going to need to be addressed further, but can be handled in a future PR.

Reply:
The reason why it's _probably_ okay in this case is that if only a single pointer is captured in a closure, it avoids a heap allocation. This is an implementation detail and there is no guarantee that this will remain the case (although I see no reason to change this).

If you want to be sure, you can try to apply #1460.